codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/middleware/MiddlewareBuilder.java

```diff
@@ -140,17 +141,39 @@ public void addDefaultMiddleware(GenerationContext context) {
                         "When `true`, does not perform host prefix injection using @endpoint's hostPrefix property.")
                 .build();
 
+        Middleware hostPrefix = (new Middleware.Builder())
+                .klass("Hearth::Middleware::HostPrefix")
+                .step(MiddlewareStackStep.INITIALIZE)
+                .addConfig(disableHostPrefix)
+                .operationPredicate((model, service, operation) -> operation.hasTrait(EndpointTrait.class))
+                .operationParams((ctx, operation) -> {
+                    Map<String, String> params = new HashMap<>();
+                    SmithyPattern pattern = operation.getTrait(EndpointTrait.class).get().getHostPrefix();
```

Comment:
General pattern should probably be expectTrait(TraitClass) instead of getTrait().get()

Reply:
I'll fix